Tests/UnitTests/EdgeRequestTests.swift

```diff
@@ -24,9 +24,22 @@ class EdgeRequestTests: XCTestCase {
     func testEncode_allProperties() {
         let konductorConfig = KonductorConfig(streaming: Streaming(recordSeparator: "A", lineFeed: "B"))
         let requestMetadata = RequestMetadata(konductorConfig: konductorConfig, state: nil)
-        var identityMap = IdentityMap()
-        identityMap.addItem(namespace: "email", id: "example@adobe.com")
-        let requestContext = RequestContextData(identityMap: identityMap)
+        guard let identityMapData = """
+            {
+              "email" : [
+                {
+                  "authenticationState" : "ambiguous",
+                  "id" : "example@adobe.com",
+                  "primary" : false
+                }
+              ]
+            }
```

Comment:
Same comment as above, the IdentityEdge shared state has a single property "identityMap" which contains all the identities. The test data should look like:
```
{
  "identityMap" : {
    "email" : [
      {
        "authenticationState" : "ambiguous",
        "id" : "example@adobe.com",
        "primary" : false
      }
    ]
  }
}
```

Reply:
In this test we populate `RequestContextData.identityMap` with the JSON shown in the PR. If we were to use the JSON suggested we would get a strange looking request when we flatten it:
```
XCTAssertEqual failed: ("{
    "xdm.identityMap.email[0].id" = "example@adobe.com";
}") is not equal to ("{
    "xdm.identityMap.identityMap.email[0].id" = "example@adobe.com";
}")
```

The `RequestContextData` type already has a key property `identityMap`, we then populate it with what would be the contents of the `identityMap` object. So in this test, I think this is the valid thing to do.

Reply:
Updated this. I got dynamic encoding to work.